lib/versioning/rez/index.ts

```diff
@@ -57,3 +67,317 @@
 //     "       )"
 //     "    )$"
 // ).format(version_group=version_group)
+
+// - Replace {version_group} -> ${versionGroup}
+// - Replace (?P<...>) -> (?<...>)
+// - Replace ?(...) -> \k<...>
+// - Replace single \ -> double \
+const versionGroup = '([0-9a-zA-Z_]+(?:[.-][0-9a-zA-Z_]+)*)';
```

Comment:
Please split the file as it's pretty big now. Maybe one file for regex and one for non public util function's.

Reply:
Updated. Please have another look.